unic/utils/tests/macro_tests.rs

```diff
@@ -0,0 +1,53 @@
+#[macro_use]
+extern crate unic_utils;
+
+use unic_utils::CharProperty;
+
+char_property! {
+    #[derive(Debug, PartialEq, Eq)]
+    pub enum Property {
+        /// Required
+        AbbrVariant {
+            abbr => AV,
+        }
+        /// Required
+        LongVariant {
+            long => Long_Variant,
+        }
+        /// Required
+        DisplayVariant {
+            display => "Display Variant",
+        }
+        /// Required
+        EmptyVariant {
+        }
+    }
+
+    pub mod abbr_names;
+}
+
+#[test]
+fn basic_macro_use() {
+    assert_eq!(Property::AbbrVariant, abbr_names::AV);
+
+    assert_eq!(Property::AbbrVariant.abbr_name(), Some("AV"));
+    assert_eq!(Property::LongVariant.abbr_name(), None);
+    assert_eq!(Property::DisplayVariant.abbr_name(), None);
+    assert_eq!(Property::EmptyVariant.abbr_name(), None);
+
+    assert_eq!(Property::AbbrVariant.long_name(), None);
+    assert_eq!(Property::LongVariant.long_name(), Some("Long_Variant"));
+    assert_eq!(Property::DisplayVariant.long_name(), None);
+    assert_eq!(Property::EmptyVariant.long_name(), None);
+
+    assert_eq!(format!("{}", Property::AbbrVariant), "AV");
+    assert_eq!(format!("{}", Property::LongVariant), "Long_Variant");
```

Comment:
I don't think that it is possible to transform the captured `:ident` in that manner using `macro_rules!`, otherwise I would have done it already. You could in theory do it with a procedural macro, but I've not yet familiarized myself with how to do those, stable or not.

If we do go the path of a procedural macro in order to be able to do this transformation, I would (naively) suggest making the whole thing a procedural macro for better error messages.